tests/unit/restapi/v1/test_queue.py

```diff
@@ -0,0 +1,516 @@
+    """Test that queues can be correctly registered and retrieved using the API.
+
+    Given an authenticated user, this test validates the following sequence of actions:
+
+    - The user registers a queue named "tensorflow_cpu".
+    - The response is valid matches the expected values given the registration request.
+    - The user is able to retrieve information about the queue using the queue id.
+    """
+    name = "tensorflow_cpu"
+    description = "The first queue."
+    user_id = auth_account["user_id"]
+    group_id = auth_account["default_group_id"]
+    queue1_response = register_queue(
+        client, name=name, description=description, group_id=group_id
+    )
+    queue1_expected = queue1_response.get_json()
+    assert_queue_response_contents_matches_expectations(
+        response=queue1_expected,
+        expected_contents={
+            "name": name,
+            "description": description,
+            "user_id": user_id,
+            "group_id": group_id,
+        },
+    )
+    assert_retrieving_queue_by_id_works(
+        client, queue_id=queue1_expected["queueId"], expected=queue1_expected
+    )
+
+
+@pytest.mark.v1
+def test_queue_get_all(
+    client: FlaskClient,
+    db: SQLAlchemy,
+    auth_account: dict[str, Any],
+    registered_queues: dict[str, Any],
+) -> None:
+    """Test that all queues can be retrieved.
+
+    Given an authenticated user and registered queues, this test validates the following
+        sequence of actions:
+
+    - A user registers three queues, "tensorflow_cpu", "tensorflow_gpu", "pytorch_cpu".
+    - The user is able to retrieve a list of all registered queues.
+    - The returned list of queues matches the full list of registered queues.
+    """
+    queue1_expected = registered_queues["queue1"].get_json()
+    queue2_expected = registered_queues["queue2"].get_json()
+    queue3_expected = registered_queues["queue3"].get_json()
+    queue_expected_list = [queue1_expected, queue2_expected, queue3_expected]
+
+    assert_retrieving_all_queues_works(client, expected=queue_expected_list)
+
+
+@pytest.mark.v1
+def test_queue_search_query(
+    client: FlaskClient,
+    db: SQLAlchemy,
+    auth_account: dict[str, Any],
+    registered_queues: dict[str, Any],
+) -> None:
+    """Test that queues can be queried with a search term.
+
+    Given an authenticated user and registered queues, this test validates the following
+        sequence of actions:
+
+    - The user is able to retrieve a list of all registered queues with a description
+      that contains 'queue'.
+    - The returned list of queues matches the expected matches from the query.
+    """
+    queue1_expected = registered_queues["queue1"].get_json()
+    queue2_expected = registered_queues["queue2"].get_json()
+    queue_expected_list = [queue1_expected, queue2_expected]
+
+    # TODO the query field must be updated when the grammar for searches is updated.
+    search_parameters = {"query": "description:*queue*"}
```

Comment:
I recommend updating the assert functions so we don't have to remember to create a dict with the `"query"` key for the `Q=` argument. We should be able to just pass the string `"description:*queue*"` since there can only be one instance of a query parameter per request.